s = "this is My First Python programming class and i am learNING python string and its function"

# 1 . Try to extract data from index one to index 30 with a jump of 3 

In [2]:
s = "this is My First Python programming class and i am learNING python string and its function"

In [9]:
s[1:30:3]

'h   r tnrr'

# 2. Try to reverse a string without using reverse function 


In [13]:
len(s)

90

In [22]:
s[-1:-91:-1]

'noitcnuf sti dna gnirts nohtyp GNINrael ma i dna ssalc gnimmargorp nohtyP tsriF yM si siht'

# 3. Try to split a string after conversion of entire string in uppercase 


In [26]:
m = s.upper()
m


'THIS IS MY FIRST PYTHON PROGRAMMING CLASS AND I AM LEARNING PYTHON STRING AND ITS FUNCTION'

In [32]:
m.split(" ")

['THIS',
 'IS',
 'MY',
 'FIRST',
 'PYTHON',
 'PROGRAMMING',
 'CLASS',
 'AND',
 'I',
 'AM',
 'LEARNING',
 'PYTHON',
 'STRING',
 'AND',
 'ITS',
 'FUNCTION']

# 4. try to convert the whole string into lower case 


In [33]:
s.lower()

'this is my first python programming class and i am learning python string and its function'

# 5 . Try to capitalize the whole string 


In [39]:
s.capitalize()

'This is my first python programming class and i am learning python string and its function'

# 6 . Write a diference between isalnum() and isalpha()

isalnum() shows the result for an alphanumeric character whereas it is isalpha() checks for an alphabetic character.

If we are running isalpha() function, then in order to have the "True" value returned, strings have to have "letters" in that. In other words, If we are willing to check the presence of letters in the string by running this function, it will return "True" in case of alphabets and "false" otherwise.  

Likewise, Python isnumeric() function returns "true" if all characters in a string are numbers and "false" in case of absence of numbers. 


# 7. Try to give an example of expand tab


In [50]:
s = "xyzyu\t12345\tabc"

In [52]:
s.expandtabs(6)

'xyzyu 12345 abc'

Explanation: When you set the tabsize to 6. The tab stops are 6, 12, 18 and so on. For 'xyzyu', the tab stop is 6 and for '12345', the tab stop is 12. Hence, there is 1 space after 'xyzyu' and 1 space after '12345'.

# 8 . Give an example of strip , lstrip and rstrip 


In [79]:
s = "          this is My First Python programming class and i am learNING python string and its function000000000000     "

In [80]:
s.strip()

'this is My First Python programming class and i am learNING python string and its function000000000000'

In [81]:
s.lstrip()

'this is My First Python programming class and i am learNING python string and its function000000000000     '

In [82]:
s.rstrip()

'          this is My First Python programming class and i am learNING python string and its function000000000000'

# 9.  Replace a string charecter by another charector by taking your own example 


In [83]:
s = "Hey!its me Megs"

In [96]:
s.replace("m", "J")

'Hey!Nts Je Megs'

In [97]:
index = 4


In [99]:
new_Character = "Y"

In [101]:
s[:index] + new_Character + s[index+1:]


'Hey!Yts me Megs'

# 10 . Try  to give a defination of string center function with and exmple 


The center string()center aligns the string, using a specified character as the fill character. We Denote fill character as "fillchar"



In [103]:
m = "Meghakanjwani"

In [106]:
m.center(30)

'        Meghakanjwani         '

In [107]:
m.center(25, "H")

'HHHHHHMeghakanjwaniHHHHHH'

# 11. Write your own definition of compiler and interpretor without copy paste form internet in your own language

Computer programs exist in high-level languages that a human can understand. They can be in the form of phrases and words. Humans can not understand the machine language, likewise computers don't understand these languages as we do. The input that we give, gets comprehended in binary codes for the computer to understand. 

As a result, we first write a program in a high-level language (source code), convert them into machine language, and make them readable for the computers. Hence, we need interpreters and compilers. The compiler creates the program by analyzing all sorts of the language statements to check if they are correct given by the user. If it finds any error in between, it gives an error message. If there are no errors spotted, the compiler will convert the source code into machine code. 

On the other side, the interpreter reads each statement of code and then executes it. They translates only one statement at a time of the program. They also create an exe of the programming language before the program runs.

# 12. Python is a interpreted of compiled language give a clear ans with your understanding 

In various books of python programming and articles, it is mentioned that python language is interpreted. But as far as one's view matters on this topic, I have found that it is first compiled and then interpreted.The compilation part is hidden from the programmer thus, many programmers believe that it is an interpreted language. The compilation part is done first when we execute our code and this will generate byte code and internally this byte code gets converted by the python virtual machine(p.v.m). Compiled part gets deleted by the python as soon as we execute the code to refrain the programmers from getting into complexity. The edit-test-debug cycle is incredibly fast hence Debugging Python programs is easier than other languages. 

# 13 . Try to write a usecase of python with your understanding .


1. Insurance
Top use: Creating business insights with machine learning

Case study: One American multinational finance and insurance corporation faced competition from smaller companies that were introducing services driven by machine learning. To compete, the insurer allowed teams to develop new applications and services using machine learning; however, with too many sets of data science tools involved, a number of different versions of Python and compatibility issues arose. The company settled on one version of Python to deliver all of the machine learning capabilities needed.

2. Retail banking
Top use: Flexible data transformation and manipulation

Case study: A large American department store chain with an in-store banking arm collects data centrally in a warehouse, and then shares it with multiple applications to enable its supply chain, retail banking, and analytics and reporting needs. While the company standardized on Python for data manipulation, each team created its own version, which created problems. The company decided on a single, standard Python build to increase engineering speed and decrease support costs.

3. Aerospace
Top use: Meeting software system deadlines

Case study: An American multinational aerospace, military, and defense corporation was contracted to provide a number of systems for the International Space Station. While aerospace software focused on critical safety systems is typically written in a language like Ada, those older languages do not lend themselves well to scripting tasks, GUI creation, or data science analysis. Selecting a single Python version offered a larger contract value and no exposure.

4. Finance
Top use: Data mining identify cross-sell opportunities

Case study: An American multinational financial services corporation wanted to mine complex customer and prospect behavioral data as part of a digital transformation project. The company used Python to initiate different data science and machine learning initiatives to examine the structured data it had been collecting for years, and correlated it with unstructured data from the web and social media to increase cross-selling and reclaim resources.

5. Business services
Top use: API access to financial information

Case study: A privately-held financial data and media company had previously provided partners with access to financial information through different electronic resources. Partners wanted to build desktop applications in a variety of languages, including Python, to incorporate the customer’s API directly into their own, and created a Python Software Development Kit (SDK) for their financial information API, leading to increased revenue and customer satisfaction.